### PR TITLE
TLUNIZIN-548 send email when there is SIS upload warnings

### PR DESCRIPTION
**Pull request by zqian**
Parse the upload result json, and find whether there is a "processing_warnings" element value. Include those values in the alert email message.

### CODE REVIEW

File: SIS_import/sis_upload.rb [hunk 1]
```diff
@@ -476,11 +513,20 @@ def create_instructor_sandbox_site(zip_file_name, outputFile)
...
-	`echo #{upload_error} | mail -s "#{$server} Upload Error" #{ENV['MAILTO']}`
+	`echo #{upload_error} | mail -s "#{$server} SIS Upload Error" #{ENV['MAILTO']}`
 else
+	if ($upload_warnings != "")
+		# mail the upload warning message
+		## check first about the environment variable setting for MAILTO '
+		p "Use the environment variable 'MAILTO' for sending out SIS upload warning messages to #{ENV['MAILTO']}"
+		p $upload_warnings
+		## send email to support team with the error message
+		`echo #{$upload_warnings}  | mail -s "#{$server} SIS Upload Warnings" #{ENV['MAILTO']}`
```
Comment (dlhaines) [inline SIS_import/sis_upload.rb]: What happens if the mail fails?  Can the status code be checked?

Comment (zqian) [inline SIS_import/sis_upload.rb]: The upload warning message also gets logged into log files. 

What "status code" do you refer here? Status code for email or status code for SIS upload?

Comment (dlhaines) [inline SIS_import/sis_upload.rb]: I was referring to the status code for mail. It's nice to know if mail has
a problem but if the information is also in the log it's probably not
necessary to start reporting on the status of mail.

File: SIS_import/sis_upload.rb [hunk 2]
```diff
@@ -167,7 +169,42 @@ def upload_to_canvas(fileName, outputFile, output_file_base_name)
...
+			#                                     "abstract_courses"=>0,
+			#                                     "courses"=>488,
+			#                                     "sections"=>488,
+			#                                     "xlists"=>0,
+			#                                     "users"=>606,
+			#                                     "enrollments"=>1178,
+			#                                     "groups"=>0,
+			#                                     "group_memberships"=>0,
+			#                                    "grade_publishing_results"=>0}},
+			#                 "batch_mode"=>null,
```
Comment (pushyamig) [inline SIS_import/sis_upload.rb]: Why is commented code is not deleted

Comment (zqian) [inline SIS_import/sis_upload.rb]: This code serves as an example for returned result data structure.

**Comment by pushyamig** [comment]
Looks Good

**Comment by dovek** [comment]
I have nothing to add beyond what Pushyami and David said except for this: why was 10 seconds decided on the sleep time?

**Comment by zqian** [comment]
10 seconds is the default value for sleep time. In fact, this value is read from the configuration file, see line 47.

**Pull request merged**